docs/how_tos/simulating_circuits_with_obp.ipynb

```diff
@@ -0,0 +1,539 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "26d926db-2e2f-41ff-947e-9757a9f426d8",
+   "metadata": {},
+   "source": [
+    "# Simulating circuits with OBP"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7642d6db-fe6d-4506-8e49-6c840f166e2b",
+   "metadata": {},
+   "source": [
+    "In this guide, you will learn how to simulate `QuantumCircuit` instances entirely through the means of OBP.\n",
+    "\n",
+    "Since OBP will take an observable and backpropagate it through a given circuit, the \"simulation\" of a circuit amounts to computing the expectation value of the target observable with respect to this circuit.\n",
+    "As you will see later, the `qiskit-addon-obp` package is even capable of handling simple noise models, allowing you to compute noisy expectation values, too!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e996d79-36fd-4a71-bd6b-737fc8e44987",
+   "metadata": {},
+   "source": [
+    "## Constructing an example circuit\n",
+    "\n",
+    "For the purposes of this guide, we will use the same example circuit as in the [Pauli term truncation guide](https://qiskit.github.io/qiskit-addon-obp/how_tos/truncate_operator_terms.html):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "b7946767-904b-422e-b076-7f956f3fdb70",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import rustworkx.generators\n",
+    "from qiskit.synthesis import LieTrotter\n",
+    "from qiskit_addon_utils.problem_generators import (\n",
+    "    PauliOrderStrategy,\n",
+    "    generate_time_evolution_circuit,\n",
+    "    generate_xyz_hamiltonian,\n",
+    ")\n",
+    "from qiskit_addon_utils.slicing import combine_slices, slice_by_gate_types\n",
+    "\n",
+    "# we generate a linear chain of 10 qubits\n",
+    "num_qubits = 10\n",
+    "linear_chain = rustworkx.generators.path_graph(num_qubits)\n",
+    "\n",
+    "# we use an arbitrary XY model\n",
+    "hamiltonian = generate_xyz_hamiltonian(\n",
+    "    linear_chain,\n",
+    "    coupling_constants=(0.05, 0.02, 0.0),\n",
+    "    ext_magnetic_field=(0.02, 0.08, 0.0),\n",
+    "    pauli_order_strategy=PauliOrderStrategy.InteractionThenColor,\n",
+    ")\n",
+    "# we evolve for some time\n",
+    "circuit = generate_time_evolution_circuit(hamiltonian, synthesis=LieTrotter(reps=3), time=2.0)\n",
+    "# slice the circuit by gate type\n",
+    "slices = slice_by_gate_types(circuit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "63bcf4d0-7abe-4467-bc2b-f95d8d66dee5",
+   "metadata": {},
+   "source": [
+    "However, the above is purely the circuit describing the time evolution under a chosen Hamiltonian.\n",
+    "We also need an initial state to start from, with respect to which we compute the expectation values of our observable.\n",
```

Comment:
This evolution circuit does have a trivial initial state though, the all-zero state, right?

Reply:
Kind of yes. But I would be careful since it does not include actual `reset` operations (although, to be fair, neither does the other `initial_state` circuit below, nor do we need them in this context).